docs/001_develop/03_client-capabilities/021_desktop-interoperability/002_fdc3-channel-event.mdx

```diff
@@ -0,0 +1,117 @@
+---
+title: 'FDC3 Channel Event'
+sidebar_label: 'FDC3 Channel Event'
+id: fdc3-channel-event
+keywords: [interaction, interactive, interoperability, openfin, interop.io, fdc3, fdc3 channel, fdc3 context, data]
+tags:
+- interaction
+- interactive
+- interoperability
+- openfin
+- interop.io
+- fdc3
+- fdc3 channel
+- fdc3 context
+- data
+sidebar_position: 1
+---
+
+# FDC3 Channel Event
+
+A utility element to that listens to an event on its parent element and broadcasts the event detail payload on an FDC3 Channel.
+
+Use cases:
+* Send event on system color channel
+* Send event on app channel
+* Map the event payload before broadcasting on a channel
+
+## Example 1 - Send event on system channel
+
+This will listen to the `rowClicked` event that is fired by the `rapid-grid-pro` element whenever a row is clicked.
+
+It will broadcast on the current system color channel.
+
+It will broadcast with the context type `fdc3.position`.
+
+The context payload will contain the `event.detail.data` payload in the `id` property of the Context object.
+
+```html
+<rapid-grid-pro>
+    <fdc3-channel-event event-name="rowClicked" channel-type="fdc3.position">
+
+    </fdc3-channel-event>
+</rapid-grid-pro>
+```
+
+## Example 2 - Send event on app channel
+
+This will listen to the `rowClicked` event that is fired by the `rapid-grid-pro` element whenever a row is clicked.
+
+It will broadcast on the specified app channel, in this case `appChannel`.
+
+```html
+<rapid-grid-pro>
+    <fdc3-channel-event event-name="rowClicked" channel-type="fdc3.position" channel-name="appChannel">
+
+    </fdc3-channel-event>
+</rapid-grid-pro>
+```
+
+## Example 3 - Send mapped event on system channel
+
+This will listen to the `rowClicked` event that is fired by the `rapid-grid-pro` element whenever a row is clicked.
+
+It will broadcast on the current system color channel.
+
+It will broadcast with the context type `fdc3.position`.
+
+The mapping function will map the `event.detail.data` payload and create a new object..
+
+```html
+<rapid-grid-pro>
+    <fdc3-channel-event
+        :mappingFunction="${() => ({TICKER}) => ({ id: { ticker: TICKER, cusip: CUSIP } })}"
+        event-name="rowClicked"
+        channel-type="fdc3.position">
+    </fdc3-channel-event>
+</rapid-grid-pro>
+```
+
```

Comment:
Happy to say no but is it worth having a comprehensive example that is enough that the user could copy it into their app and have the full flow? Correct me if I am wrong, but having just `<rapid-grid-pro>` without any datasource etc specific isn't quite enough to be a full example

Reply:
Added the grid datasource (that points to the entity example I included)